csm_web/scheduler/admin.py

```diff
@@ -235,13 +227,100 @@ def get_students(self, obj):
     get_students.short_description = "Students"
 
 
+class SectionForm(forms.ModelForm):
+
+    def __init__(self, *args, **kwargs):
+        user = kwargs.pop("user")
+        super().__init__(*args, **kwargs)
+        queryset = Section.objects.filter(course__in=get_visible_courses(user))
+        self.queryset = (
+            queryset
+            .select_related("mentor__user")
+            .prefetch_related("students")
+        )
+        fields = self.fields
+        # Prevent inappropriate foreign key editing
+        fields["course"].widget.can_change_related = False
+        fields["course"].widget.can_add_related = False
+        fields["mentor"].disabled = True
+        fields["mentor"].help_text = "The mentor for this section. To change, please click the adjacent edit or add buttons."
+        if self.instance.pk is not None:
+            spacetime = self.instance.spacetime
+            has_mentor = self.instance.mentor is not None
+            fields["mentor_name"].initial = self.instance.mentor.name if has_mentor else "-"
+            fields["mentor_email"].initial = self.instance.mentor.user.email if has_mentor else "-"
+            fields["mentor_profile_id"].initial = self.instance.mentor.pk if has_mentor else "-"
+            fields["location"].initial = spacetime._location
+            fields["start_time"].initial = spacetime._start_time
+            fields["duration"].initial = spacetime._duration
+            fields["day_of_week"].initial = spacetime._day_of_week
+            fields["students"].initial = "\n".join(str(student) for student in self.instance.students.all())
+        else:
+            for field in ("mentor_name", "mentor_email", "mentor_profile_id", "students"):
+                fields[field].widget = forms.HiddenInput()
+
+    mentor_name = forms.CharField(required=False, disabled=True)
+    mentor_email = forms.CharField(required=False, disabled=True)
+    mentor_profile_id = forms.CharField(required=False, disabled=True)
+    spacetime = forms.ModelChoiceField(required=False, queryset=Spacetime.objects.all(), widget=forms.HiddenInput())
+    location = forms.CharField(max_length=100)
+    start_time = forms.TimeField(
+        widget=forms.TimeInput(attrs={'type': 'time'}),
+        help_text="Enter a value of the form 'hh:mm', e.g. '13:00' for a 1PM section."
+    )
+    duration = forms.DurationField(
+        widget=forms.TimeInput(attrs={'type': 'time'}),
```

Comment:
Duration widget should **not** be set to time, this makes it display with AM/PM, which makes no sense for a duration